### PR TITLE
Pipeline with failover to multi cluster

### LINKED ISSUE #3568: Add support for automatic failover in Pipeline
### Expected behavior

The `Pipeline` class supports automatic failover.

### Actual behavior

The `Pipeline` class uses the `Connection` class directly instead of utilizing the Command executor with Circuit breaker support.

### Steps to reproduce:

Please create a reproducible case of your problem. Make sure
that case repeats consistently and it's not random
1. Follow the https://github.com/redis/jedis/blob/master/docs/failover.md instructions
2. Use `pipelined()` to send commands as a command
3. Simulate the Redis connection problem by shutting down the first cluster.

### Redis / Jedis Configuration
https://github.com/redis/jedis/blob/master/docs/failover.md

#### Jedis version:
5.0

#### Redis version:
Any

#### Java version:
Any

### PR DESCRIPTION
**Pull request by sazzad16**
Resolves #3568

### CODE REVIEW

File: src/main/java/redis/clients/jedis/UnifiedJedis.java [hunk 1]
```diff
@@ -4822,8 +4822,11 @@ public Object tFunctionCallAsync(String library, String function, List<String> k
...
       throw new IllegalStateException("It is not allowed to create Pipeline from this " + getClass());
+    } else if (provider instanceof MultiClusterPooledConnectionProvider) {
+      return new MultiClusterFailoverPipeline((MultiClusterPooledConnectionProvider) provider);
+    } else {
+      return new Pipeline(provider.getConnection(), true);
     }
-    return new Pipeline(provider.getConnection(), true);
   }
 
   public Transaction multi() {
```
Comment (banker) [inline src/main/java/redis/clients/jedis/UnifiedJedis.java:4832]: @sazzad16 See my comment in https://github.com/redis/jedis/issues/3568#issuecomment-1750799211.

This is also a problems for multi(), subscribe(), and any methods that use JedisCommandIterationBase. Shouldn't we fix these as well?

**Comment by banker** [comment]
We need to ensure that we've tested these changes. Would you like assistance with that @sazzad16 ?

**Comment by sazzad16** [comment]
> We need to ensure that we've tested these changes.

@banker I haven't found tests from our original failover implementation to imitate here for pipeline implementation. The only closest thing I can see is https://github.com/redis/jedis/blob/d2f6712cb633a5cba84ef0c555be7caeec7a8abb/src/test/java/redis/clients/jedis/providers/MultiClusterPooledConnectionProviderTest.java#L102-L114

File: src/main/java/redis/clients/jedis/UnifiedJedis.java (hunk 1 above)
Comment (chayim) [reply src/main/java/redis/clients/jedis/UnifiedJedis.java:4840]: Current plan: Those will be in a different issue, and estimated there.

**Review by chayim — APPROVED** [review]

File: src/main/java/redis/clients/jedis/UnifiedJedis.java (hunk 1 above)
Comment (chayim) [reply src/main/java/redis/clients/jedis/UnifiedJedis.java:4840]: #3602 contains multi and supersedes this subscribe

**Pull request closed without merging**